Purchases/Attribution/RCAttributionTypeFactory.m

```diff
@@ -1,78 +0,0 @@
-//
-// Created by Andrés Boedo on 2/25/21.
-// Copyright (c) 2021 Purchases. All rights reserved.
-//
-
-#import "RCAttributionTypeFactory.h"
-
-NS_ASSUME_NONNULL_BEGIN
-
-@implementation RCAttributionTypeFactory
-
-- (Class<FakeAdClient> _Nullable)adClientClass {
-    return (Class<FakeAdClient> _Nullable)NSClassFromString(@"ADClient");
-}
-
-- (NSString *)mangledIdentifierClassName {
-    return @"NFVqragvsvreZnantre";
-}
-
-- (NSString *)mangledIdentifierPropertyName {
-    return @"nqiregvfvatVqragvsvre";
-}
-
-- (NSString *)mangledAuthStatusPropertyName {
-    return @"genpxvatNhgubevmngvbaFgnghf";
-}
-
-- (NSString *)mangledTrackingClassName {
-    return @"NGGenpxvatZnantre";
-}
-
-- (Class<FakeTrackingManager> _Nullable)atTrackingClass {
-    // We need to do this mangling to avoid Kid apps being rejected for getting idfa.
-    // It looks like during the app review process Apple does some string matching looking for
-    // functions in ATTrackingTransparency. We apply rot13 on these functions and classes names
-    // so that Apple can't find them during the review, but we can still access them on runtime.
-    NSString *className = [self rot13:self.mangledTrackingClassName];
-
-    return (Class<FakeTrackingManager> _Nullable)NSClassFromString(className);
```

Comment:
this line from the old code may seem straightforward, but what it's actually doing is the tricky part - it's relying on the fact that `FakeTrackingManager` exposes the same methods we're looking for in `ATTrackingManager` to call the same methods. 
Since in objc this really means passing a message with the same name and values, it'll work as long as the interface of `FakeTrackingManager` matches whatever we're trying to use from `ATTrackingManager`

Reply:
I was intrigued by this, and having spent a good deal of time mucking around with the objc runtime, I wanted to give it a go with Swift. Turns out, it's not too bad! Just not as clever as what currently exists:

Since we're working with Apple's ObjC classes, that means we still have NSObject subclasses that we're dealing with. So, we can do some dynamic dispatch 🕺

```swift
guard let klass = NSClassFromString("ASIdentifierManager"), let myClass = klass as? NSObjectProtocol else {
    // ASIdentifierManager hasn't been loaded in the ObjC runtime yet
    print("sad panda")
    return
}

let sharedManager = myClass.perform(Selector("sharedManager")).takeRetainedValue()
let uuid = sharedManager.perform(Selector("advertisingIdentifier")).takeRetainedValue()

```

Reply:
![Screen Shot 2021-07-09 at 6 29 17 PM](https://user-images.githubusercontent.com/1304821/125147917-93650000-e0e3-11eb-8987-1183ff8051ed.png)

Reply:
great idea! @Juanpe I think this approach would work. We'd still need to retain the ROT13 mangling, of course. 

@taquitos could you clarify the bit about referencing the class in code? Currently this works even if the class isn't referenced, as long as the framework has been linked in the build phases. 
I think if you don't link it, then this only works if you've referenced the class, a behavior that started in iOS 14.something (can't recall the exact release). 

Could this be what you're seeing?

Reply:
I'm 99% sure that's exactly it- it must be linked. Last I knew, linking happens when a class is referenced at least once in the codebase. That is, the compiler hasn't optimized it out. This is a problem for ObjC where a framework includes categories, but the app using the framework doesn't use the category. The compiler would strip it out unless you used the `-objc` linker flag to retain those things. In the past it was not enough to have the library/framework linked in the xcode project, it needed to be referenced somewhere in the app's code.

Reply:
The good news, I think this is relatively easy to check for- build an app and run the code.